flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -3596,14 +3596,34 @@ class ArrayExprLowering {
           operands[arg.firArgument] = [=](IterSpace iters) { return exv; };
         }
       } break;
-      case PassBy::CharBoxValueAttribute:
-        TODO(loc, "CHARACTER, VALUE");
-        break;
-      case PassBy::BoxChar:
-        TODO(loc, "CHARACTER");
-        break;
+      case PassBy::CharBoxValueAttribute: {
+        if (isArray(*expr)) {
+          PushSemantics(ConstituentSemantics::RefOpaque);
+          auto lambda = genarr(*expr);
+          operands[arg.firArgument] = [=](IterSpace iters) {
+            return Fortran::lower::CharacterExprHelper{builder, loc}
+                .createTempFrom(lambda(iters));
+          };
+        } else {
+          Fortran::lower::CharacterExprHelper helper(builder, loc);
+          auto argVal = helper.createTempFrom(asScalarRef(*expr));
+          operands[arg.firArgument] = [=](IterSpace iters) -> ExtValue {
+            return argVal;
+          };
+        }
+      } break;
+      case PassBy::BoxChar: {
+        PushSemantics(ConstituentSemantics::RefOpaque);
+        auto lambda = genarr(*expr);
+        operands[arg.firArgument] = [=](IterSpace iters) {
+          return lambda(iters);
+        };
```

Comment:
It doesn't look like you need the lambda wrapping `lambda` with this one.

Reply:
Good point, I cleaned that up here as well as in a few other spots just above.